base/src/main/scala/data/ProCompose.scala

```diff
@@ -0,0 +1,54 @@
+package scalaz
+package data 
+
+object ProCompose {
+  import typeclass.{Profunctor, Strong, Category, StrongClass, CategoryClass}
+  import typeclass.Strong._
+  import typeclass.Profunctor._
+
+  trait ProCompose[P[_, _], A, B] { } 
+
+  case class Pure[P[_, _], A, B](pab: A => B) extends ProCompose[P, A, B]
+
+  case class Composed[P[_, _], A, X, B](pab: P[A, X], more: ProCompose[P, X, B]) extends ProCompose[P, A, B]
```

Comment:
👎  for `Vector`, because:
 - not type aligned, thus would require `asInstanceOf`s;
 - a naive catenable will have cheaper `cons`, `snoc`, `++`, random access is not required, and IMO _O(1)_ `uncons` is not really necessary either, only that _n_ successive `uncons` will be _O(n)_.

It's easy to change the constructors so that a naive catenable is basically baked in.